skbio/stats/tests/test_composition.py

```diff
@@ -269,6 +355,136 @@ def test_centralize(self):
                             np.array([[2, 2, 6],
                                       [4, 4, 2]]))
 
```

Comment:
Can you add the following test cases?

- input that is too small for ANCOM (not enough samples and/or features)
- groups of size 1 (i.e., each sample in its own group)
- a single group
- `table` and `grouping` that have same samples but different order should give same results as a test where they're ordered the same
- noncontiguous groups in `grouping` should give same result as contiguous groups
- specifying different `tau`, `alpha`, and `theta` values that are valid